question22/main.js

```diff
@@ -0,0 +1,188 @@
+  },
+  age: {
+    value : "年齢",
+    sort : true
+  }
+}
+
+const renderTableElement = (userData)  => {
+  const table = createElementWithClassName("table", "table");
+  const tbody = createElementWithClassName("tbody", "tbody");
+  table.id = "js-table";
+  tbody.id = "js-tbody";
+  wrapper.appendChild(table).appendChild(getTableHeadElement());
+  wrapper.appendChild(table).appendChild(tbody).appendChild(getTableRowFragment(userData));
+  sortButtonClickEvent(userData);
+}
+
+const getTableHeadElement = () => {
+  const thead = createElementWithClassName("thead", "thead");
+  const tr = createElementWithClassName("tr", "tr");
+  const fragmentThElement = document.createDocumentFragment();
+  const userDataColumnKeys = Object.keys(userDataColumn);
+
+  userDataColumnKeys.forEach((key) => {
+    const th = createElementWithClassName("th", "th");
+    const value = userDataColumn[key].value;
+    th.textContent = value; 
+    userDataColumn[key].sort && th.appendChild(createSortButton());
+    fragmentThElement.appendChild(tr).appendChild(th); 
+  });
+
+  thead.appendChild(fragmentThElement);
+  return thead;
+}
+
+const getTableRowFragment = (userData) => {
+  const tableRowfragment = document.createDocumentFragment();
+  for (let i = 0; i < userData.length; i++) {
+    const tr = createElementWithClassName("tr", "tr");
+    tableRowfragment.appendChild(tr).appendChild(getTableDataFragment(userData[i]));
+  }
+  return tableRowfragment;
+}
+
+const getTableDataFragment = (userData) => {
+  const tableDatafragment = document.createDocumentFragment();
+  Object.keys(userDataColumn).forEach((key) => {
+    const td = createElementWithClassName("td", "td");
+    td.textContent = userData[key];
+    tableDatafragment.appendChild(td);
+  });
+  return tableDatafragment;
+};
+
+const createSortButton = () => {
+  const sortButton = createElementWithClassName("button", "sortButton");
+  sortButton.dataset.status = "default";
+  sortButton.id = "js-sortbtn";
+  return sortButton;
+}
+
+const sortButtonClickEvent = () => {
+  const tbody = document.querySelector('tbody');
+  const sortButtons = [...document.querySelectorAll(".sortButton")];
+  const trElement = [...document.querySelectorAll("tbody > tr")];
+  sortButtons.forEach((sortButton)=> {
+    sortButton.addEventListener('click', (e) => {
+      resetSortbuttonStatus(sortButtons, e.target);
+      const clickedCellIndex = e.target.parentElement.cellIndex;
+      const nextStatus = switchSortStatus(e.target);
+      sortButton.dataset.status = nextStatus;
+      const sortedRows = sortFunc(e.target,clickedCellIndex,trElement);
+      while (tbody.firstChild) {
+        tbody.removeChild(tbody.firstChild);
+    }
+      sortedRows.forEach((row) => {
+        tbody.appendChild(row);
+      });
+    });
+  })
+}
+
+const resetSortbuttonStatus = (sortButtons, target) => {
+  sortButtons.forEach((button)=> {
+    if (button !== target) {
+      button.dataset.status = Sort.Default;
+    }
+  })
+}
```

Comment:
If one of the buttons is pressed, the status of the other button is reset to default.

Reply:
IMO
I thought it would be easier to understand if I was more specific, what do you think?

```suggestion
const resetSortbuttonStatus = (sortButtons, clickedButton) => {
  sortButtons.forEach((sortButton) => {
    if (sortButton === clickedButton) return;
    sortButton.dataset.status = Sort.Default;
  });
};
```

Reply:
Thank you for your review! Yes it is true!!  I have fixed your idea as is.
fixed commit : https://github.com/csakyo/homework/pull/37/commits/bb33c29e377e542eafc41cc40197accf196b9e31